pkg/action/executor/k8s_controller.go

```diff
@@ -176,3 +169,67 @@ func (c *parentController) shouldSkipOperator(controller *unstructured.Unstructu
 func (rc *parentController) String() string {
 	return rc.name
 }
+
+//To determine if ORM should be used or not in terms of the owner reference of the work load controller
+func (c *parentController) shouldUseOrm(controller *unstructured.Unstructured) bool {
+	if controller == nil {
+		glog.V(3).Infof("Workload controller pointer is nil, shouldUseOrm return false")
+		return false
+	}
+	controllerOwnerReferences := c.obj.GetOwnerReferences()
+	if len(controllerOwnerReferences) <= 0 {
+		glog.V(3).Infof("Workload controller <%v/%v> doesn't have an owner reference, shouldUseOrm return false", controller.GetNamespace(), controller.GetName())
+		return false
+	}
+	/*
+		In the following 3 situations, we use trigger ORM to update the CR
+	*/
+	//Case 1: The owner of the controller has the kind as ClusterServiceVersion
+	/*
+		ownerReferences:
+		- apiVersion: operators.coreos.com/v1alpha1
+		  blockOwnerDeletion: false
+		  controller: false
+		  kind: ClusterServiceVersion
+		  name: ibm-licensing-operator.v1.14.0
+		  uid: bfd3d22d-99f9-473d-bea6-dd81b8d763ff
+		resourceVersion: "438731373"
+		uid: 6e58b1e7-dd29-41d5-8b77-194797b4fe9e
+	*/
+	if "ClusterServiceVersion" == controllerOwnerReferences[0].Kind {
```

Comment:
Could controllerOwnerReferences[0].Kind  be nil?

Reply:
the type of the controllerOwnerReferences[0].Kind  is string, it will never be nil.